adapter/service/virtualmeter/manager.go

```diff
@@ -484,14 +481,16 @@ func (m *managerWrapper) registerVirtualServices(
 
 	srv := numericmeter.NewService(publisher, &numericmeter.Config{
 		Specification:     numericSpec,
-		Reporter:          newController(topic, m.manager),
+		Reporter:          newController(topic, m),
 		ReportingStrategy: nil,
 	})
 
 	log.Debugf("manager: registering a service template, topic: %s", topic)
 
 	device, err := m.storage.Device(topic)
 
+	log.Infof("I am updating device : %v", device)
```

Comment:
The same as above, either remove or add proper log structure. Also instead of dumping the object just refer to some identification maybe?